rpmio/rpmkeyring.c

```diff
@@ -319,10 +319,16 @@ rpmRC rpmKeyringVerifySig(rpmKeyring keyring, pgpDigParams sig, DIGEST_CTX ctx)
 	    rpmlog(rc ? RPMLOG_ERR : RPMLOG_WARNING, "%s\n", lints);
 	    free(lints);
 	}
-    }
-
-    if (keyring)
+	if (keyptr)
+	    *keyptr = key;
```

Comment:
Does this leak `key` if `keyptr` is `NULL`?

Reply:
The other way round. Should be fixed now.